.github/workflows/brs-commit-checker.yml

```diff
@@ -0,0 +1,93 @@
+                type: string
+                required: true
+                description: The git ref end of comparison range. Must be descendant of `from_git_ref`. Prefix branches with `origin/`.
+            # Jira user name & API token is used for processing sensitive tickets
+            jira_user_email:
+                type: string
+                required: false
+                description: (optional to include sensitive tickets) Email address of Jira user
+            # Jira user name & API token is used for processing sensitive tickets
+            # We protect the API token as a secret by preventing its value from pasted at the command line
+            # by using add-mask to obfuscate secret values in the jobs
+            # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions
+            jira_user_api_token:
+                type: string
+                required: false
+                description: (optional to include sensitive tickets) API token of Jira user
+
+jobs:
+    commit-report:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                fetch-tags: true
+                fetch-depth: 0
+            # workaround for bug in checkout action. this step should be redundant.
+            # https://github.com/actions/checkout/issues/1471
+            # https://github.com/actions/checkout/issues/1781
+            # https://github.com/actions/checkout/issues/701#issuecomment-1133937950
+            - name: Fetch all tags
+              run: |
+                git fetch --tags origin
+            - name: Fetch all branches
+              run: |
+                git fetch origin
+            - name: Setup Python
+              uses: actions/setup-python@v5
+              with:
+                python-version: '3.12.8'
+                cache: 'pipenv'
+                cache-dependency-path: |
+                    tools/commit-checker/Pipfile
+                    tools/commit-checker/Pipfile.lock
+            - name: Install pipenv
+              run: |
+                sudo pip3 install pipenv
+              # Use add-mask to obfuscate secret values in the jobs
+              # https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions
+              # https://stackoverflow.com/a/71698374/2077918
+            - name: Store Jira credentials if provided
+              run: |
+                JIRA_USERNAME=$(cat $GITHUB_EVENT_PATH | jq -r '.inputs.jira_user_email' )
+                echo "::add-mask::${JIRA_USERNAME}"
+                JIRA_PASSWORD=$(cat $GITHUB_EVENT_PATH | jq -r '.inputs.jira_user_api_token' )
+                echo "::add-mask::${JIRA_PASSWORD}"
+                if [ ! -z "${JIRA_USERNAME}" ] && [ ! -z "${JIRA_PASSWORD}" ]
+                then
+                    echo "::add-mask::${JIRA_USERNAME}"
```

Comment:
Sorry, not an expert, but this has a bit of "code smell"
When working with GitHub secrets there is a lot of masking done automatically.

I would have to dig a bit for best practices.

Reply:
See the links that I pasted in the comments above this step. Within the first link, see [this section about `add-mask`](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#masking-a-value-in-a-log)

Reply:
I think I wrote the script such that environment variables for the secrets would work. I use a dotenv file locally but you don't need to in CI if there is a better option

Reply:
I didn't see instructions about environment variables, But if that's possible, that would make using the Jira API tokens in the GHA workflows even safer via GH tokens.